readthedocs/search/tasks.py

```diff
@@ -212,3 +213,27 @@ def record_search_query(project_slug, version_slug, query, total_results, time_s
     )
     obj.created = time
     obj.save()
+
+
+@app.task(queue='web')
+def increase_page_view_count(project_slug, version_slug, path):
+    today_date = timezone.now().date()
+    page_view_obj = PageView.objects.filter(
+        project__slug=project_slug,
+        version__slug=version_slug,
+        path=path,
+        date=today_date,
+    ).first()
+
+    if page_view_obj:
+        page_view_obj.view_count = F('view_count') + 1
+        page_view_obj.save()
+    else:
+        project = Project.objects.get(slug=project_slug)
+        version = Version.objects.get(project=project, slug=version_slug)
+        PageView.objects.create(
+            project=project,
+            version=version,
+            path=path,
+            date=today_date,
+        )
```

Comment:
This whole function could be simplified to something like this:

```python
# This assumes that the default `view_count` is 0 instead of 1 as it is currently
page_view, _ = PageView.objects.get_or_create(
    # If we passed project/version IDs to this function instead of slugs, we could save 2 queries too!
    project=Project.objects.get(slug=project_slug),
    version=Version.objects.get(project=project, slug=version_slug),
    path=path,
    date=timezone.now().date(),
)
PageView.objects.filter(pk=page_view.pk).update(
    view_count=F('view_count') + 1
)
```